static/src/javascripts/projects/common/modules/identity/modules/show-errors.js

```diff
@@ -0,0 +1,40 @@
+// @flow
+import fastdom from 'lib/fastdom-promise';
+
+const formErrorClassName = 'form__error';
+const formErrorHolderClassName = 'js-errorHolder';
+
+const genericErrorMessage = 'Sorry, something went wrong';
+
+export const push = (error: any, action: string = 'none'): Promise<void> =>
```

Comment:
Is there a way to define a type for `error`? Would `Error` work? Using `any` is not recommended as it disables all type checking, including implicit type checking, for this argument.

Reply:
I can just provide `Error`  and it is indeed desirable to send errors HOWEVER show-errors has to interact with reqwest which sometimes (going offline for instance) will not return errors but an xmlhttprequest, so for the time being i just set it as any so tests can run for the "wrong" types too and see if it recovers gracefully from that

Reply:
I don't think it matters, since `reqwest` doesn't have any meaningful type annotations anyway. Do you get Flow compiler errors if you use `Error`?

Reply:
If it's possible it might accept `Error` or `XMLHttpRequest`, you could specify a [union type](https://flow.org/en/docs/types/unions/#toc-union-type-syntax):
```js
error: Error | XMLHttpRequest
```

Reply:
the overarching issue is more that show-errors might interact with _anything_ and get inputted _anything_ from a potential number of places rather than this particular case so it really feels like it should be able to take any type. I could enforce `Error` as it is the only desired type and just let it do its thing but it feels like lying

Reply:
In which case, we can accept a `mixed` type, and then [refine](https://flow.org/en/docs/types/mixed/#toc-anything-goes-in-nothing-comes-out):
```js
const push = (error: mixed ... ) => {
   errorEl.innerHTML = `<p>${error instanceof Error 
     ? error.message 
     : genericErrorMessage}. </p>`;
}
```

Reply:
aaa that's awesome! ^^